misc/install-elfutils.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+#-*- mode: shell-script; -*-
+
+prefix=/usr/local
+
+objdir=$(readlink -f ${objdir:-${PWD}})
+builddir=${objdir}/.build
+
+n_cpus=$(cat /proc/cpuinfo | grep ^processor | wc -l)
+
+usage() {
+    echo "Usage: $0 [<options>]
+
+  --help             print this message
+  --prefix=<DIR>     set install root dir as <DIR>        (default: /usr/local)
+
+  Example usage for host compilation:
+    $ $0 --prefix=./build.host
+
+  Example usage for cross compilation:
+    $ CROSS_COMPILE=arm-linux-gnueabi- ARCH=arm CFLAGS=\"-march=armv7-a -mfloat-abi=softfp\" \\
+        $0 --prefix=./build.arm
+"
+    exit 1
+}
+
+while getopts ":ho:-:p" opt; do
+    case "$opt" in
+        -)
+	    # process --long-options
+	    case "$OPTARG" in
+                help)  usage ;;
+                *=*)   opt=${OPTARG%%=*}; val=${OPTARG#*=}
+                       eval "${opt/-/_}='$val'" ;;
+                *)     ;;
+            esac
+	    ;;
+        *)       usage ;;
+    esac
+done
+shift $((OPTIND - 1))
+
+mkdir -p ${builddir} && cd ${builddir}
+
+ELFUTILS_VERSION=0.164
+ELFUTILS_NAME=elfutils-$ELFUTILS_VERSION
+ELFUTILS_TARBALL=$ELFUTILS_NAME.tar.bz2
+ELFUTILS_URL=https://sourceware.org/elfutils/ftp/$ELFUTILS_VERSION/$ELFUTILS_TARBALL
+
+wget -c $ELFUTILS_URL
+tar xfj $ELFUTILS_TARBALL
+rm -rf elfutils
+mv $ELFUTILS_NAME elfutils
+cd elfutils
+
+opt_host_cc=""
+if [ ! -z $CROSS_COMPILE ]; then
+    HOST=$(basename $CROSS_COMPILE | sed 's/-$//g')
+    opt_host_cc="--host=$HOST CC=${CROSS_COMPILE}gcc"
+fi
+
+./configure --prefix=$prefix $opt_host_cc
```

Comment:
Likewise, please do this if Makefile is not there.

Reply:
We have to check whether configure option is same as before, but I just made it simply check whether Makefile is there or not.

Reply:
I just modified it only when configure parameters are the same.